sklearn/ensemble/tests/test_stacking.py

```diff
@@ -646,6 +653,90 @@ def fit(self, X, y):
         stacker.n_features_in_
 
 
+@pytest.mark.parametrize("stack_method", ["auto", "predict", "predict_proba"])
+@pytest.mark.parametrize("passthrough", [False, True])
+def test_stacking_classifier_multilabel(stack_method, passthrough):
+    X_train, X_test, y_train, y_test = train_test_split(
+        X_multilabel_r, y_multilabel_r, stratify=y_multilabel_r, random_state=42
+    )
```

Comment:
Rather than creating a test for both problem, you should be able to parametrize this to-be-renamed test on pairs of datasets as follows:

```suggestion
@pytest.mark.parametrize("passthrough", [False, True])
@pytest.mark.parametrize(
    "X, y",
    [(X_multilabel_r, y_multilabel_r), (X_binary, y_binary)],
    ids=["multilabel", "binary"],
)
def test_stacking_classifier(stack_method, passthrough, X, y):
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, stratify=y, random_state=42
    )
```

Potentially, the current logic for the array shapes' assertions might not be changed.

Note: this formatting might not be `black`-compliant.

Reply:
Yes, I thought about parametrizing either `make_dataset` functions or some `X_y` tuples, but I thought the many twists on assertions weren't worth the effort.
Otherwise, wrapping the fitting of the model into a single function for both tests might be effective, but I wonder if that is the best practice.
I will go ahead and apply your suggestion then.
Which potential black formating issue are you referring to?

Reply:
Okay it's working, I added a boolean flag to separate the binary from the multilabel case, WDYT? I thought about using the `ids` kwargs instead but [it looks a bit ugly](https://stackoverflow.com/questions/56466111/how-to-find-the-test-id-for-parametrized-tests-during-execution-of-the-test).
Also, which potential black formating issue are you referring to?

Reply:
> Okay it's working, I added a boolean flag to separate the binary from the multilabel case, WDYT? I thought about using the ids kwargs instead but it looks a bit ugly.

I think it's a right approach. :+1:

Could `if` could be grouped together to ease reading?

>  Also, which potential black formating issue are you referring to?

Potentially an improper formatting (e.g. no new lines needed) of the parametrization I was proposing -- but it looks properly formatted (since the CI is/was not yelling).

Reply:
Done :)

Reply:
Thanks!